cmake/checks/check_01_cxx_features.cmake

```diff
@@ -122,10 +118,28 @@ IF(NOT DEFINED DEAL_II_WITH_CXX14 OR DEAL_II_WITH_CXX14)
       "
       DEAL_II_HAVE_CXX14_MAKE_UNIQUE)
 
+    #
+    # Clang-3.5* or older, bail out with a spurious error message in case
+    # of an undeduced auto return type.
+    #
+    # https://llvm.org/bugs/show_bug.cgi?id=16876
```

Comment:
Thanks. Of course the comment you had here before was no worse than what was there before, but we may still try to make things better over time.